### PR TITLE
(win10mct) Migration from DarwinJS Repo

### PR DESCRIPTION
**Pull request by RedBaron2**
@gep13 @AdmiringWorm @pascalberger @majkinetor 
## Description

This is to migrate this package with a AU update file to the core team repository.

## Motivation and Context
Why is this change required? What problem does it solve?
This makes this package a AU updating package

## How Has this Been Tested?

Include details of of your testing environment, and the tests your ran to
Updates via AU Streams version in [AppVeyor](https://ci.appveyor.com/project/RBZ/development/build/1.0.369)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Migrated package (a package have been migrated from another repository)

## Checklist:

- [X] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this mean usually the notes in the description of a package).
- [X] I have updated the documentation accordingly (this mean usually the notes in the description of a package).
- [X] All files is up to date with the latest [Contributing Guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md)
- [ ] The added/modified package passed install/uninstall in the chocolatey test environment.
- [ ] The changes only affect a single package (not including meta package).

## Original Location
- [Original Repository](add_link_to_original_repository_location)
- [Open Issues](link_to_the_generic_location_of_open_issues) *Add the different issues underneath, and tick those that are fixed in this PR*
  - [ ] Issue 1 link
  - [ ] Issue 2 Link
- [ ] *Include the link to the opened PR that removes the package from the original location*
- [X] The [migration guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Package-migration-process) have been followed
[Migrating Link](https://github.com/DarwinJS/ChocoPackages/pull/55)

### CODE REVIEW

**Comment by RedBaron2** [comment]
Adjustments in development branch, updating files soon
should it be a new PR or just adjusted here?

**Comment by RedBaron2** [comment]
Updated and AppVeyor Check passed 😄

**Comment by RedBaron2** [comment]
@majkinetor The requested changes have been made the files. 
The nuspec url(s) commit will be fixed by AU ( I Hope).

**Comment by majkinetor** [comment]
Not everything is fixed:
- Mandatory tags
- You download the file 2 times - once to get version, and again to get hashsum. Disable automatic checksum and calculate the `$Latest.ChecksumXY` yourself from downloaded file.

**Comment by AdmiringWorm** [comment]
@RedBaron2 just make the update to the mandatory tags, I plan to make some changes to the update script myself (mostly to disable the need to download the binary when it isn't needed).

**Comment by AdmiringWorm** [comment]
@majkinetor mind having a new look, I've updated his PR with quite a few changes (including the mandatory tag)

File: automatic/windows10-media-creation-tool/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# <img src="https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/tree/master/icons/windows10-media-creation-tool.png" width="48" height="48"/> [windows10-media-creation-tool](https://chocolatey.org/packages/windows10-media-creation-tool)
```
Comment (majkinetor) [inline automatic/windows10-media-creation-tool/README.md]: Readme should be sorted out. Some of text belongs to Notes. Features if possible.

File: automatic/windows10-media-creation-tool/tools/chocolateyInstall.ps1 [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+$packageArgs = @{
+  PackageName  = $env:chocolateyinstall
+  FileFullPath = "$toolsDir\$exeName"
+  Url          = ''
+  Checksum     = ''
+  ChecksumType = ''
+}
+Get-ChocolateyWebFile @packageArgs
+
+$AppPathKey = "Registry::HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\$($exeName)"
```
Comment (majkinetor) [inline automatic/windows10-media-creation-tool/tools/chocolateyInstall.ps1]: This entire section should be replaced with `Register-Application` from core-extension.

File: automatic/windows10-media-creation-tool/tools/chocolateyInstall.ps1 [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+If (!(Test-Path $AppPathKey)) {New-Item "$AppPathKey" | Out-Null}
+Set-ItemProperty -Path $AppPathKey -Name "(Default)" -Value "$($packageArgs.FileFullPath)" -Force
+Set-ItemProperty -Path $AppPathKey -Name "Path" -Value "$toolsDir" -Force
+
+Write-Output "******************************************************************************************************************"
```
Comment (majkinetor) [inline automatic/windows10-media-creation-tool/tools/chocolateyInstall.ps1]: I would remove this entire section to be consistent with other packages. This is basic OS usage, its equally valuable to inform users that [x] should be clicked to close the window.

File: automatic/windows10-media-creation-tool/windows10-media-creation-tool.nuspec [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    <projectUrl>https://www.microsoft.com/en-us/software-download/windows10</projectUrl>
+    <iconUrl>https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/tree/master/icons/windows10-media-creation-tool.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Create a bootable ISO or Flashdrive to apply the free Windows 10 upgrade with a clean install (from scratch install).
+    Does not support Windows 10 Enterprise.
+    It is also possible to install Windows 10 on an existing machine using this tool.
+    Allows a clean installation of Windows using the free upgrade.  Expand sections until you find "Perform a clean installation using USB or DVD" on this page: https://www.microsoft.com/en-us/software-download/windows10</description>
+    <summary>Create a bootable ISO or Flashdrive to apply the free Windows 10 upgrade with a clean install (from scratch install).</summary>
+    <copyright>Microsoft</copyright>
+    <tags>portable windows10-media-creation-tool freeware</tags>
```
Comment (majkinetor) [inline automatic/windows10-media-creation-tool/windows10-media-creation-tool.nuspec]: Package name is poor tag. Perhaps `windows10 media` should replace it.

File: automatic/windows10-media-creation-tool/README.md (hunk 1 above)
Comment (RedBaron2) [reply automatic/windows10-media-creation-tool/README.md]: added notes and features

**Comment by RedBaron2** [comment]
@AdmiringWorm 
Thanks for the assistance. This might help another package. 

@majkinetor 
Changes made but might need to rename the directory if the package name is to be changed to `win10mct`?

Unless anymore obvious changes need to be made this seems to be ready 🤞

**Comment by majkinetor** [comment]
You miss dependency on core extension now.

**Comment by majkinetor** [comment]
Yes, change the directory if the package gets renamed. Also the icon name

**Comment by RedBaron2** [comment]
I can take out the meta package since it is the failure issue. I created it as a fall back point for those of the older package should they want to upgrade via Chocolatey. It works on the principle of the javaruntime package.

Let me know what else needs to be done. 😄

**Pull request merged**